### PR TITLE
LPS-68813

### PR DESCRIPTION
**Pull request by Ithildir**
Hi @topolik!

Can you please take a look at this and see if it's okay now?

Also, is there a way to pass directories instead of the individual files to FindBugs? Running it on Windows is a problem, the command line is definitely too long 😞

In order to try it, go to `modules/sdk/gradle-plugins-defaults` and run `../../../gradlew install updateFileVersions`.

### CODE REVIEW

**Comment by Ithildir** [comment]
ci:stop

**Comment by topolik** [comment]
Hi @Ithildir,

I went through FindBugs codebase.

Version 3 comes with `-analyzeFromFile` option where we can specify filename to read classes from.

Seems FB supports directories and jar files.

Also there is `-xargs` option which tells FindBugs to read classes from stdin ... https://github.com/findbugsproject/findbugs/blob/master/findbugs/src/java/edu/umd/cs/findbugs/TextUICommandLine.java#L725-L740

However, this won't help us with `-sourcepath` and `-auxclasspath` which are another sources of potential complications. So we need this feature too: https://sourceforge.net/p/findbugs/bugs/1375/

For `-auxclasspath` there is a new switch in 3.0 `-auxclasspathFromFile` - read the dependencies from file.

----
Another option - looks like FindBugs supports something called `Project` which can be saved and loaded from a file - https://github.com/findbugsproject/findbugs/blob/master/findbugs/src/java/edu/umd/cs/findbugs/Project.java#L704

This file can be specified using `-project` option ... https://github.com/findbugsproject/findbugs/blob/master/findbugs/src/java/edu/umd/cs/findbugs/FindBugsCommandLine.java#L61

So we could write everything into XML file and offer that to FB.

----

In the end I have a question. Wouldn't it be easier to just patch our gradle with https://github.com/gradle/gradle/pull/781, use the standard `findBugsMain` gradle task and then upgrade to newer gradle when they merge it?

Also, I'd like to transfer the ownership to @stian-sigvartsen, so CCing him here.

Thanks.

File: modules/sdk/gradle-plugins-defaults/src/main/java/com/liferay/gradle/plugins/defaults/internal/LiferayCIPlugin.java [hunk 1]
```diff
@@ -124,7 +124,7 @@ private void _configureTaskExecuteNpm(
 
 	private void _configureTaskNpmInstall(NpmInstallTask npmInstallTask) {
 		npmInstallTask.setNodeModulesCacheDir(_NODE_MODULES_CACHE_DIR);
-		npmInstallTask.setRemoveShrinkwrappedUrls(true);
+		npmInstallTask.setRemoveShrinkwrappedUrls(Boolean.TRUE);
```
Comment (topolik) [inline modules/sdk/gradle-plugins-defaults/src/main/java/com/liferay/gradle/plugins/defaults/internal/LiferayCIPlugin.java]: I guess this is not related :)

Comment (Ithildir) [inline modules/sdk/gradle-plugins-defaults/src/main/java/com/liferay/gradle/plugins/defaults/internal/LiferayCIPlugin.java]: Yeah, not at all! 😄

**Comment by Ithildir** [comment]
Hi @topolik!
Can you have a look at this and tell me if it's okay? I used a `-project` argument to pass the aux classpath, the classpath and the source dirs.

**Comment by topolik** [comment]
Pull request submitted at: https://github.com/Ithildir/liferay-portal/pull/551

View just my changes: https://github.com/topolik/liferay-portal/compare/Ithildir:f954312621...dbe902a027
View total diff: https://github.com/topolik/liferay-portal/compare/03ba9c34ee...dbe902a027
